plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFEInputSpent.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../PaymentExitDataModel.sol";
+import "../PaymentInFlightExitModelUtils.sol";
+import "../routers/PaymentInFlightExitRouterArgs.sol";
+import "../spendingConditions/IPaymentSpendingCondition.sol";
+import "../spendingConditions/PaymentSpendingConditionRegistry.sol";
+import "../../utils/ExitId.sol";
+import "../../../utils/UtxoPosLib.sol";
+import "../../../utils/Merkle.sol";
+import "../../../vaults/EthVault.sol";
+import "../../../framework/PlasmaFramework.sol";
+import "../../../transactions/PaymentTransactionModel.sol";
+
+library PaymentChallengeIFEInputSpent {
+    using UtxoPosLib for UtxoPosLib.UtxoPos;
+    using PaymentInFlightExitModelUtils for PaymentExitDataModel.InFlightExit;
+
+    // TODO: Use BondSize lib.
+    uint256 public constant PIGGYBACK_BOND = 31415926535 wei;
+
+    struct Controller {
+        PlasmaFramework framework;
+        PaymentSpendingConditionRegistry spendingConditionRegistry;
+        uint256 supportedTxType;
+        EthVault ethVault;
+    }
+
+    event InFlightExitInputBlocked(
+        address indexed challenger,
+        bytes32 txHash,
+        uint8 inputIndex
+    );
+
+    function run(
```

Comment:
Just noticed when I am doing new spending condition interface for challenge SE.
(I think) We need to check `isProtocolFinalized()` whenever calling spending condition. This is new and previously no action need to be done due to every tx is MoreVP....

Reply:
@kevsul I think you missed to handle this?

Reply:
Ah, I did miss it 🤦‍♂ 
Will do